vm/systemSmartContracts/liquidStaking.go

```diff
@@ -0,0 +1,244 @@
+//go:generate protoc -I=proto -I=$GOPATH/src -I=$GOPATH/src/github.com/ElrondNetwork/protobuf/protobuf  --gogoslick_out=. liquidStaking.proto
+package systemSmartContracts
+
+import (
+	"bytes"
+	"fmt"
+	"sync"
+
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	"github.com/ElrondNetwork/elrond-go-core/core/atomic"
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
+	"github.com/ElrondNetwork/elrond-go-core/hashing"
+	"github.com/ElrondNetwork/elrond-go-core/marshal"
+	"github.com/ElrondNetwork/elrond-go/config"
+	"github.com/ElrondNetwork/elrond-go/vm"
+	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
+)
+
+const tokenIDKey = "tokenID"
+const noncePrefix = "n"
+
+type liquidStaking struct {
+	eei                      vm.SystemEI
+	sigVerifier              vm.MessageSignVerifier
+	delegationMgrSCAddress   []byte
+	liquidStakingSCAddress   []byte
+	endOfEpochAddr           []byte
+	gasCost                  vm.GasCost
+	marshalizer              marshal.Marshalizer
+	hasher                   hashing.Hasher
+	mutExecution             sync.RWMutex
+	liquidStakingEnableEpoch uint32
+	flagLiquidStaking        atomic.Flag
+}
+
+// ArgsNewLiquidStaking defines the arguments to create the liquid staking smart contract
+type ArgsNewLiquidStaking struct {
```

Comment:
according to our last talk about naming convention, this should be `LiquidStakingArgs`